deepchem/models/torch_models/lcnn.py

```diff
@@ -0,0 +1,428 @@
+    self.UseBN = UseBN
+    self.activation = Shifted_softplus()
+    self.dropout = nn.Dropout(p=dropout)
+
+  def forward(self, node_feats):
+    """
+        Update node representations.
+
+        Parameters
+        ----------
+        node_feats: torch.Tensor
+            The node features. The shape is `(N, Node_feature_size)`.
+
+        Returns
+        -------
+        node_feats: torch.Tensor
+            The updated node features. The shape is `(N, Node_feature_size)`.
+
+        """
+
+    node_feats = self.conv_weights(node_feats)
+    if self.UseBN:
+      node_feats = self.batch_norm(node_feats)
+
+    node_feats = self.activation(node_feats)
+    node_feats = self.dropout(node_feats)
+
+    return node_feats
+
+
+class Shifted_softplus(nn.Module):
+  """
+    Performs a Shifter softplus loss, which modifies with a value of log(2)
+    """
+
+  def __init__(self):
+    super(Shifted_softplus, self).__init__()
+    self.act = nn.Softplus()
+    self.shift = nn.Parameter(torch.tensor([0.69310]), False)
+
+  def forward(self, X):
+    """
+        Applies the Activation function
+
+        Parameters
+        ----------
+        node_feats: torch.Tensor
+            The node features.
+
+        Returns
+        -------
+        node_feats: torch.Tensor
+            The updated node features.
+
+        """
+    node_feats = self.act(X) - self.shift
+    return node_feats
+
+
+class Custom_dropout(nn.Module):
+  """
+    An implementation for few , Given a task perform a rowise sum of 2-d
+    matrix , you get a zero out the contribution of few of rows in the matrix
+    Given, X a 2-d matrix consisting of row vectors (1-d) x1 , x2 ,..xn.
+    Sum = x1 + 0.x2 + .. + 0.xi + .. +xn
+    """
+
+  def __init__(self, dp_rate: float, n_permutation: int):
+    """
+        Parameters
+        ----------
+        dp_rate: float
+            p value of dropout.
+        """
+    super(Custom_dropout, self).__init__()
+    self.dropout = nn.Dropout(p=dp_rate)
+    self.ones = nn.Parameter(torch.ones(n_permutation), requires_grad=False)
+
+  def forward(self, layer):
+    """
+        Returns
+        -------
+        node_feats: torch.Tensor
+            Updated tensor.
+        """
+    mask = self.dropout(self.ones).view(layer.shape[0], 1).repeat(
+        1, layer.shape[1])
+    return mask * layer
+
+
+class LCNN(nn.Module):
+  """
+    The Lattice Convolution Neural Network (LCNN)
+
+    This model takes lattice representation of Adsorbate Surface to predict
+    coverage effects taking into consideration the adjacent elements interaction
+    energies.
+
+    The model follows the following steps
+
+    [1] It performs n lattice convolution operations.
+        For more details look at the LCNNBlock class
+    [2] Followed by Linear layer transforming into sitewise_n_feature
+    [3] Transformation to scalar value for each node.
+    [4] Average of properties per each element in a configuration
+
```

Comment:
Can you add a usage example section to this docstring that shows how to use this model?